adds orbs_allowlist and test cases

eddiewebb opened the pull request:
## Rationale

We see more customers interested in allowing specific 3rd party orbs than banning known bad ones.  Chatted in slack with David about new helper.

## Considerations

Why did you make some of the technical decisions you did in implementing this PR?
- modeled after existing functions and test.

Were any other approaches considered and rejected?
- Right now allow list is just coordinates, not version.  I think versioning might be used in the existing ban.

(allow orbs A, B, C.  Ban older version A:1.2)

## Changes

- add `orbs_allowlist` rule in internal/rego/orbs.rego
- add test cases in policies/helpers/orbs/allowlist

davidmdm commented on internal/helpers/rego/config/orbs.rego:
```diff
@@ -16,4 +16,11 @@ ban_orbs_version(banned_orbs) = { orb: msg | orb := banned_orbs[_]
\ No newline at end of file
+}
+
+orbs_allowlist(allowed_orbs) = { orb: msg | orb := input["orbs"][_]
```
I am not sure this helper is accomplishing what we said in the slack thread.

As far as I can tell it:
- has a guard to check if there are orbs, otherwise it early exits
- Looks for the orb value name but not key:
```yaml
orbs:
  key: name@version
```
- then if the name is not in allowed list of names it returns an error

This isn't very secure because all it does is allow a list of orbs but it doesn't specify how they must be namespaced "keyed" nor does it accept multiple versions of an orb.

What is preferable from a customer perspective?
```
config.orbs_allowlist({"good", "safe", "ok"})
```
Such that you can only import orbs with those names?

OR

Police the keys to fully qualified orb sets?

```
config.orb_allowlist({
  "good": "good@1.0.0",
  "ok": {"ns/ok@1.5.0", "ns/ok@2.0.0",
})
```

eddiewebb replied:
I explicity left off version,  but disagree on the name/key issue, and have a test to measure that explicity.

Orbs are targeted by `namespace/name:version` and this rule makes sure `namespace\name` is allowed, regardless of the local key they assign it.

Any orbs defined with  a namespace./ name not allowed will fail.

eddiewebb replied:
We dont care what devs call it within their config, as long as we approve its use.

davidmdm replied:
Ok... I have just found that often we want to make sure that orbs are assigned to a known key so that we can enforce their execution within steps. Although we can decide that it is best left to the specific policy that enforces certain commands to enforce the orb key to ns+name+version...

Its a locality versus centralization problem

davidmdm commented on cpa/tester/policies/helpers/orbs/allowlist/policy.rego:
```diff
@@ -0,0 +1,14 @@
+
+policy_name["orbs_allowlist"]
+
+enable_rule["orbs_allowlist"] {
```
have you not heard of `enable_hard`? It should be released in the docs as of today :) and has been live for a week or two

eddiewebb commented:
@davidmdm  - thanks for review/approval.

I changed test policy to use enable_hard and squashed my 5 commits into 1. 

Since I don't have write access I can't merge this myself.
